### PR TITLE
feat: new lesson chip

### PR DESCRIPTION
**Pull request by 01zulfi**
## Because

testing if adding a lesson chip is viable

## This PR

adds "new" lesson chip

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [ ] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
    - `Feature` - adds new or amends existing user-facing behavior
    - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
    - `Fix` - bug fixes
-   [ ] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] I have verified all tests and linters pass after making these changes.
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

**Comment by CouchofTomato** [comment]
@01zulfi 

Do you need anyone else to do anything with this mate?

**Comment by 01zulfi** [comment]
@CouchofTomato not sure, I can take this to completion. Will need help if there any other changes required

File: app/models/lesson.rb [hunk 1]
```diff
@@ -20,6 +20,11 @@ class Lesson < ApplicationRecord
 
   attribute :completed, :boolean, default: false
 
+  def new?
```
Comment (KevinMulhern) [inline app/models/lesson.rb:23]: Nit: `new?` is a little too close to the method `new` that most ruby classes have. It'll work fine, but might cause a little bit of confusion and require a second glance when reading through the code.

We could go with something more expressive here:

```suggestion
  def recently_added?
```

or 
```suggestion
  def recently_created?
```

Comment (KevinMulhern) [inline app/models/lesson.rb:23]: We'll need a unit test for this method.

Something very similar to what we've got for the like `created_today` scope should do the job:
https://github.com/TheOdinProject/theodinproject/blob/main/spec/models/like_spec.rb#L12

We'll just need to test the edges
- Created today
- Just within the new period - 1.month.ago
- Just outside of the new period - 1.month.ago + 1.day

**Review by KevinMulhern — COMMENTED** [review]
This is looking ace @01zulfi, nice work! Just a few small suggestions from me.

One problem we'll have, is every lesson will be marked as "new" on fresh installations of the app locally and on review apps. But I'm not sure if that really is a problem, and I can't think of a reasonable way around it. We might just have to document it if contributors are confused by it.

File: app/models/lesson.rb [hunk 2]
```diff
@@ -20,6 +20,11 @@ class Lesson < ApplicationRecord
...
+    true
+    # (Time.zone.now - created_at).to_i < 5.day
```
[the remark this answers is not in the document]
Comment (01zulfi) [reply app/models/lesson.rb]: Initially, I was planning anywhere from 5-7 days, but now that you have said 1 month, I can't help but increase this period. Settling for 2 weeks for now.

File: spec/models/lesson_spec.rb [hunk 3]
```diff
@@ -118,6 +118,29 @@
     end
   end
 
+  describe '#recently_added?' do
+    context 'when the lesson is added today' do
```
Comment (KevinMulhern) [inline spec/models/lesson_spec.rb:122]: Nit: _Extremely_ minor, but using a consistent tense, usually past tense, makes the test readability flow a tiny bit better.

```suggestion
    context 'when the lesson was added today' do
```

File: spec/models/lesson_spec.rb [hunk 4]
```diff
@@ -118,6 +118,29 @@
...
+      it 'returns true' do
+        lesson = create(:lesson, created_at: Time.zone.today)
+        expect(lesson.recently_added?).to be(true)
+      end
+    end
+
+    context 'when the lesson is added within last 2 weeks' do
```
Comment (KevinMulhern) [inline spec/models/lesson_spec.rb:129]:
```suggestion
    context 'when the lesson was added within last 2 weeks' do
```

File: app/models/lesson.rb [hunk 5]
```diff
@@ -20,6 +20,10 @@ class Lesson < ApplicationRecord
 
   attribute :completed, :boolean, default: false
 
+  def recently_added?
+    created_at > 2.weeks.ago.beginning_of_day
```
Comment (KevinMulhern) [inline app/models/lesson.rb:24]: `beginning_of_day` nice touch! 🤌

File: app/views/lessons/_header.html.erb [hunk 6]
```diff
@@ -3,9 +3,16 @@
...
 
     <div class="text-center xl:text-left">
-      <h1 class="text-2xl font-semibold text-gray-800 dark:text-gray-200 min-w-full" data-test-id="lesson-title-header">
-        <%= lesson.display_title %>
-      </h1>
+      <div class="flex items-center gap-3">
+        <h1 class="text-2xl font-semibold text-gray-800 dark:text-gray-200" data-test-id="lesson-title-header">
+          <%= lesson.display_title %>
+        </h1>
+        <% if lesson.recently_added? %>
```
Comment (KevinMulhern) [inline app/views/lessons/_header.html.erb:10]: Nit: Seeing a little bit of weirdness on smaller screens. We could maybe break the badge to a new line on the sm breakpoint and below

Screenshot 2024-06-11 at 23 54 02

Comment (01zulfi) [reply app/views/lessons/_header.html.erb:10]: nice catch!

Comment (KevinMulhern) [reply app/views/lessons/_header.html.erb:10]: Thanks @01zulfi - looks perfect

**Review by KevinMulhern — APPROVED** [review]
🔥

**Pull request merged**